### DISCUSSION TITLE
For those who use Apples – Potentially useful patch for the Makefile

### DISCUSSION THREAD

**Discussion by Gandoon** [XEphem/XEphem#74] [General]
Hi all,
It has been some time since I looked in on this project. 
However, I thought I might say a few words on how it works on a modern M1 Macbook Pro.
It works like a charm. I built it against X, OpenSSL, and OpenMotif from Macports, which I have done previously on an Intel machine running MacOS 10.15 with the same result. 

There are a few things to keep in mind when building to make the build find your OpenSSL and OpenMotif though. 
I thought I may offer a diff that can be used for patching if you too are building using MacPorts as your backbone. I have mildly updated my recommendation I previously made in #23 (I do not use the "static linking" section this time). 
If you are using a different package system (such as fink or homebrew) or none at all, you may need to further edit the paths to point to where your libraries live.

I have also fixed the botched output from the manpage generation. Using groff (also from MacPorts in my case), the addition of the postprocessor directive `-P-c` gives the proper output when the manpage is piped to a file. Without it, the output would look ok in a terminal directly, but not when stored in a file as was done originally. This way it looks the way it should when asking `man xephem`.

With this patch, a simple `make` will actually do just what it says on the tin. Produce a fresh xephem binary that works. I also opted to use clang-15 instead of gcc, just like last time I mentioned this here, it tended to produce a minutely smaller binary and, as far as I can tell, no unwanted behaviour such as crashes.

As previously noted, only .txt-files are allowed here, so I had to add the ".txt" to the filename.

[Makefile_MacPorts.diff.txt](https://github.com/XEphem/XEphem/files/10579737/Makefile_MacPorts.diff.txt)

**Comment by willyg5us** [comment]
Success!  I've used Gandoon's instructions to get XEphem to build and run.

I managed to get xephem to build on an M1 Mac mini with xephem 4.1.0, also using MacPorts, for MacOS Ventura 13.5.1.  I did have to alter two of the lines Gandoon altered in his patch file to match my own environment.  Here's the only changes for my environment:

< around line 36>

ifeq ($(UNAME_S),Darwin)
  PLATI = -I/opt/local/libexec/openssl3/include/openssl -I/opt/local/include/X11
  PLATL = -L/opt/local/libexec/openssl3/include/openssl -L/opt/local/include/X11
endif

Thanks for getting this working!
